jwst/datamodels/container.py

```diff
@@ -181,9 +186,20 @@ def from_asn(self, asn_data, asn_file_path=None):
         asn_file_path: str
             Filepath of the association, if known.
         """
-        # make a list of all the input files
-        infiles = [member['expname'] for member
-                   in asn_data['products'][0]['members']]
+        # match the allowed_exptypes to the exptype in the association and retain
+        # only those file that match, as a list, if allowed_exptypes is set to none
+        # grab all the files
+        if self.allowed_exptypes:
+            infiles = []
+            print("Filtering datasets based on allowed exptypes:", self.allowed_exptypes)
```

Comment:
Design: No `print` statements. These should be logging. I would guess at the `INFO` level.

Reply:
Agree on logging vs. print, but I'm not sure we'd want this at the `INFO` level. We already get complaints from regular users about too much infrastructure and process flow control type info filling up the logs. So perhaps `DEBUG`.

Reply:
Set output as logging a the debug level as per Howard's request.